site/docs-md/guides/push-notifications-firebase.md

```diff
@@ -0,0 +1,369 @@
+
+- Your **iOS bundle ID** should match the **appId** from your `capacitor.config.json` file
+- We used `com.mydomain.myappname` for this Capacitor app ID, so that is what we'll use for this entry.
+- App Nickname and App Store ID are optional
+
+Then click the **Register app** button.
+
+### Add the `GoogleService-Info.plist` file to your iOS app
+
+**Note**: This is **not** the same file used for your Android app.
+
+Download the `GoogleService-Info.plist` provided to your local machine and move it into your Xcode project as instructed, ensuring to add it to all targets. 
+
+![Google Service Info Plist Location for iOS](/assets/img/docs/guides/firebase-push-notifications/google-plist-location-ios.png)
+
+**Note**: It is recommended to use Xcode to move the `.plist` file into your project.
+
+### Add the Firebase SDK via CocoaPods
+
+The Push Notification API on iOS makes use of CocoaPods - an iOS dependency management system - and we need to tell CocoaPods to make use of Firebase.
+
+To do this, we need to modify the `Podfile`, which is located in your `ios/App` directory:
+
+![Podfile Location iOS](/assets/img/docs/guides/firebase-push-notifications/podfile-location-ios.png)
+
+We need to modify the App target pods to include Firebase. To do that, add `pod Firebase/Messaging` to your `target 'App'` section, like so:
+
+```ruby
+import Capacitor
+import Firebase # Add this line
+
+...
+
+target 'App' do
+capacitor_pods
+# Add your Pods here
+pod 'Firebase/Messaging' # Add this line
+end
+```
+
+Your `Podfile` should look something like this:
+
+```ruby
+platform :ios, '11.0'
+use_frameworks!
+
+# workaround to avoid Xcode 10 caching of Pods that requires
+# Product -> Clean Build Folder after new Cordova plugins installed
+# Requires CocoaPods 1.6 or newer
+install! 'cocoapods', :disable_input_output_paths => true
+
+def capacitor_pods
+  # Automatic Capacitor Pod dependencies, do not delete
+  pod 'Capacitor', :path => '../../node_modules/@capacitor/ios'
+  pod 'CapacitorCordova', :path => '../../node_modules/@capacitor/ios'
+  
+  # Do not delete
+end
+
+target 'App' do
+  capacitor_pods
+  # Add your Pods here
+  pod 'Firebase/Messaging'
+end
+```
+
+### Add Initialization Code
+
+To connect to Firebase when your iOS app starts up, you need to add the following initialization code to your `AppDelegate.swift` file, in the `application(didFinishLaunchingWithOptions)` method:
+
+```swift
+FirebaseApp.configure()
+```
+
+Your `AppDelegate.swift` file's `application(didFinishLaunching)` method should look something like this:
+
+```swift
+var window: UIWindow?
+
+func application(_ application: UIApplication,
+  didFinishLaunchingWithOptions launchOptions: [UIApplicationLaunchOptionsKey: Any]?) -> Bool {
+  FirebaseApp.configure()
+  return true
+}
+```
+
+### Update the Project
+
+Now we'll need to ensure that our iOS project is updated with the proper Firebase CocoaPod installed.
```

Comment:
I think it's better to do this before doing the code changes as Firebase won't be available until this is run and will cause build failures on the Xcode project